redaxo/src/core/lib/util/version.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * Validator class.
+ *
+ * @author staabm
+ *
+ * @package redaxo\core
+ */
+class rex_version
+{
+    private function __construct()
+    {
+        // noop
+    }
+
+    /**
+     * @param string|null $version
+     */
+    public static function isUnstable($version): bool
+    {
+        // see https://www.php.net/manual/en/function.version-compare.php
+        foreach (['dev', 'alpha', 'a', 'beta', 'b', 'rc', 'pl'] as $unstable) {
+            if (false !== stripos($version, $unstable)) {
+                return true;
```

Comment:
`1.0-foo` ist aktuell stable und `1.0-bar` unstable.
Ich würde entweder nur schauen, ob a-z vorkommt. Oder ansonsten dann aber richtig nach den Parts schauen, nicht nur mit `stripos`.

Reply:
'final' oder 'stable' o.ä. keywords können auch vorkommen und sind nicht unstable

Reply:
> Oder ansonsten dann aber richtig nach den Parts schauen, nicht nur mit `stripos`.

haben wir dazu irgendwelche fertig funktionen im core?

Reply:
> 'final' oder 'stable' o.ä. keywords können auch vorkommen und sind nicht unstable

Laut version_compare sind sie sogar noch kleiner als "dev".
`1.0-stable < 1.0-dev < 1.0-beta < 1.0`

> haben wir dazu irgendwelche fertig funktionen im core?

Ich würde es dann per regex machen, so in der Art:
`/(?<![a-z])(?:dev|alpha|a|beta|b|rc|pl)(?![a-z])/i`

Reply:
Mach es gerne so fertig. Bin heut nicht mehr am rechner

Reply:
Tests hinzugefügt - machst du den regex fertig?

Reply:
jupp, kann ich machen